[layout & remove & refactor & style] : 전체적인 코드 정리, 커스텀 스플래시 UI, 검색 페이지 무한 스크롤, 앱 아이콘 변경

heesoo-park opened the pull request:
## PR 체크리스트
다음 요구사항을 충족하는지 확인해주세요

- [x] 커밋메세지 규칙을 따릅니다.

## PR 유형
어떤 변경사항이 있는지 모두 체크해 주세요

- [x] 기능구현
- [ ] 버그픽스
- [x] 리팩토링
- [x] 문서 변경
- [x] 레이아웃

## 변경사항 작성
변경사항의 상세 정보를 작성해주세요.

-기능
레이아웃 xml 정리
사용하지 않는 파일 정리
주석 정리
상수값 이동
사용하지 않는 코드 제거
커스텀 스플래쉬 화면
검색 페이지 무한 스크롤 기능
앱 아이콘 변경
-설명
id 컨벤션 확인
ctrl + alt + L 로 줄 정리
필요없는 코드 삭제
제거된 파일들 : (CategoryVideoAdapter.kt, IdSearchList.kt, SavedItem.kt, SaveVideoAdapter.kt)
themes.xml에서 <item name="android:windowIsTranslucent">true</item> 을 추가하여 디폴트 스플래쉬화면이
뜨지 않도록 했고 메인 액티비티 처음에 커스텀 스플래쉬용 레이아웃을 3초 동안 띄웠다가 1.5초동안 페이드아우싱 되도록 했다.
이를 통해 데이터를 받는 동안 비어있는 홈 화면을 사용자가 보지 않도록 했다.
onScrolled에서 현재 리사이클러뷰의 8번째 아이템이라면 데이터를 불러오도록 했다.
처음에는 8번째에 있으면 무한으로 계속 불러와서 거의 몇 초만에 할당량을 초과했었는데 이 부분을 막기 위해 Boolean 변수를 추가해 8번째 아이템 도착했을 때 막고 데이터를 받아와서 리스트를 업데이트할 때 풀어주도록 했다.
다음 검색 결과를 가져오는 건 pageToken을 이용했고 처음에는 null로 세팅했다가 이후에는 넘겨주는 pageToken을 저장해놨다가 사용하는 방식으로 했다.
데이터는 잘 받아오는데 리스트가 계속 재정렬되는 문제가 있었다.
그래서 이리저리 테스트를 해보다가 전역변수로 만들어놓은 리스트 변수가 문제라는 결론에 다다랐다.
그래서 이 부분을 넘어온 데이터를 바로 가공하여 넣어보는 방식으로 변경했더니 원하는 모양의 무한 스크롤이 이루어졌다.

heesoo-park committed:
layout : 레이아웃 xml 정리

id 컨벤션 확인, ctrl + alt + L 로 줄 정리, 필요없는 코드 삭제

heesoo-park committed:
remove : 사용하지 않는 파일 정리

CategoryVideoAdapter.kt
IdSearchList.kt
SavedItem.kt
SaveVideoAdapter.kt

heesoo-park committed:
style, refactor : 주석 정리, 상수값 이동, 사용하지 않는 코드 제거

Box-o reviewed (commented):
Date or Time Formatter가 중복 사용되고 있습니다
Utils로 분리해 사용하는 것도 고려해 보면 좋을 것 같습니다 😄

Box-o reviewed (approved):
사용하지 않는 주석화 되어 있는 코드와 이미 내용공유가 끝난 부분의 설명 주석이 사라져 가독성이 좋아진 것 같습니다!!

Box-o commented:
테스트입니다!

Box-o commented:
코멘트 알림 테스트 2

Box-o commented:
알림테스트4

Box-o commented:
알림 테스트5

Box-o commented:
코드 별로네요

Box-o reviewed (approved):
승인 테스트 입니다

heesoo-park commented:
안녕하세요

Box-o reviewed (approved):
승인 테스트 두번째 입니다

Box-o reviewed (approved):
승인 테스트 세번째 입니다

heesoo-park committed:
feat : 커스텀 스플래쉬 화면 생성

themes.xml에서 <item name="android:windowIsTranslucent">true</item> 을 추가하여 디폴트 스플래쉬화면이
뜨지 않도록 했고 메인 액티비티 처음에 커스텀 스플래쉬용 레이아웃을 3초 동안 띄웠다가 1.5초동안 페이드아우싱 되도록 했다.
이를 통해 데이터를 받는 동안 비어있는 홈 화면을 사용자가 보지 않도록 했다.

heesoo-park committed:
feat : 검색 페이지 무한 스크롤 기능 추가

onScrolled에서 현재 리사이클러뷰의 8번째 아이템이라면 데이터를 불러오도록 했다.
처음에는 8번째에 있으면 무한으로 계속 불러와서 거의 몇 초만에 할당량을 초과했었는데 이 부분을 막기 위해 Boolean 변수를 추가해 8번째 아이템 도착했을 때 막고 데이터를 받아와서 리스트를 업데이트할 때 풀어주도록 했다.
다음 검색 결과를 가져오는 건 pageToken을 이용했고 처음에는 null로 세팅했다가 이후에는 넘겨주는 pageToken을 저장해놨다가 사용하는 방식으로 했다.
데이터는 잘 받아오는데 리스트가 계속 재정렬되는 문제가 있었다.
그래서 이리저리 테스트를 해보다가 전역변수로 만들어놓은 리스트 변수가 문제라는 결론에 다다랐다.
그래서 이 부분을 넘어온 데이터를 바로 가공하여 넣어보는 방식으로 변경했더니 원하는 모양의 무한 스크롤이 이루어졌다.

heesoo-park committed:
style : 앱 아이콘 변경

Box-o reviewed (approved):
스플래시 구현 화면공유로 실시간 확인했습니다

Box-o reviewed (approved):
많은 api 트래픽을 희생해서 결국 완성해 내었네요!!!!!!!!!!!!

Box-o reviewed (approved):
앱 아이콘 변경 확인했습니다
